src/sampling.jl

```diff
@@ -316,12 +343,28 @@ function sample!(a::AbstractArray, x::AbstractArray; replace::Bool=true, ordered
     return x
 end
 
+
+"""
+    sample(a, [wv::WeightVec], n::Integer; replace=true, ordered=false)
+
+Select a random, optionally weighted sample of size `n` from an array `a`
+using a polyalgorithm. `replace` dictates whether sampling is performed
+with replacement and `order` dictates whether an ordered sample, also called
+a sequential sample, should be taken.
+"""
 function sample{T}(a::AbstractArray{T}, n::Integer; replace::Bool=true, ordered::Bool=false)
-    sample!(a, Array(T, n); replace=replace, ordered=ordered)
+    sample!(a, Array{T}(n); replace=replace, ordered=ordered)
 end
 
+
+"""
+    sample(a, [wv::WeightVec], dims::Dims; replace=true, ordered=false)
+
+Select a random, optionally weighted sample from an array `a` specifying
```

Comment:
A polyalgorithm is a set of different algorithms with a rule for choosing when to use each one. 

Experimenting with the function a bit I figured out what the `dims` argument is actually doing: It fills an array of the given dimensions with samples from the input.